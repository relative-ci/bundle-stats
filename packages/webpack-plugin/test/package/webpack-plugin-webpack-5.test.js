const webpack5 = require('webpack');
const MemoryFS = require('memory-fs');
const { advanceTo } = require('jest-date-mock');
const { set } = require('lodash');

const config = require('./app/webpack.config');

advanceTo(new Date(2020, 10, 30));

jest.setTimeout(10 * 1000);

describe('webpack plugin package', () => {
  test('webpack5', (done) => {
    expect.assertions(3);

    const compiler = webpack5(config);
    compiler.outputFileSystem = new MemoryFS();

    compiler.run((error, stats) => {
      expect(error).toEqual(null);
      expect(stats.hasErrors()).toBe(false);

      // Set bundle-stats.html size to 0 to ignore size changes
      const { assets } = stats.toJson({ source: false, assets: true });
      set(assets, '[0].size', 0);
      set(assets, '[0].info.size', 0);

      expect(assets).toMatchSnapshot();
      done();
    });
  });
});
