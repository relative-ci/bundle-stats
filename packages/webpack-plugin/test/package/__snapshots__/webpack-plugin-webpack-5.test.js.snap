// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack plugin package webpack5 1`] = `
[
  {
    "auxiliaryChunkIdHints": [],
    "auxiliaryChunkNames": [],
    "auxiliaryChunks": [],
    "cached": false,
    "chunkIdHints": [],
    "chunkNames": [],
    "chunks": [],
    "comparedForEmit": false,
    "emitted": true,
    "filteredRelated": undefined,
    "info": {
      "development": true,
      "size": 0,
    },
    "isOverSizeLimit": false,
    "name": "bundle-stats.html",
    "related": {},
    "size": 0,
    "type": "asset",
  },
  {
    "auxiliaryChunkIdHints": [],
    "auxiliaryChunkNames": [],
    "auxiliaryChunks": [],
    "cached": false,
    "chunkIdHints": [],
    "chunkNames": [
      "main",
    ],
    "chunks": [
      179,
    ],
    "comparedForEmit": false,
    "emitted": true,
    "filteredRelated": undefined,
    "info": {
      "javascriptModule": false,
      "minimized": true,
      "size": 27,
    },
    "isOverSizeLimit": false,
    "name": "main.js",
    "related": {},
    "size": 27,
    "type": "asset",
  },
]
`;
